### PR TITLE
OpenAI working, new components and state usage

### PR DESCRIPTION
**Pull request by ghost**
Work in progress.
Implemented OpenAI's API, states, and a very basic UI. 

Update
Basic UI upgraded a little bit.
Tickets completed in this PR:
https://github.com/unosquare/modgen.js/issues/2
: In the previos PR i did no implementation of Zustand. This PR has one and it's in use.
https://github.com/unosquare/modgen.js/issues/3
:
Open AI implemented and running.

Did a little rework of the React Query hooks i was using.

### CODE REVIEW

**Comment by ghost** [comment]
GPT in use
PlainUI

**Comment by ghost** [comment]
Pictures above showcase the UI when you arrive to the page, and after you input something

File: src/app/inicio/page.tsx [hunk 1]
```diff
@@ -1,29 +1,71 @@
...
-import { QueryClient, QueryClientProvider, useQuery } from "react-query";
+const openai = new OpenAI({
+  apiKey: process.env.NEXT_PUBLIC_OPENAI_API_KEY, 
+  dangerouslyAllowBrowser: true
+});
 
 export default function Home() {
-  const {data, status} = useQuery('data', dataFetch);
+  //Zudstand for the input of the user
+  const { userInputs} = userInputStore();
```
Comment (AntonioGGovea) [inline src/app/inicio/page.tsx]: you can unify this by extracting both members from the same hook.
`const [userInputs, updateUserInput] = userInputStore((state) => [state.userInputs, state.updateUserInput]);`

**Review by AntonioGGovea — CHANGES_REQUESTED** [review]

File: src/app/inicio/page.tsx (hunk 1 above)
Comment (ghost) [reply src/app/inicio/page.tsx]: Great, thank you. Just implemeted it

**Review by AntonioGGovea — APPROVED** [review]

**Review by oscarfgq — APPROVED** [review]

**Pull request merged**